crates/wash-lib/src/parser/mod.rs

```diff
@@ -603,6 +611,276 @@ impl DevManifestComponentTarget {
+                false
+            }
+            // The only option left is that the versions are the same and their versions are incompatible/different
+            _ => true
+        }
+    }
+}
+
+impl std::str::FromStr for WitInterfaceSpec {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match parse_wit_package_name(s) {
+            Ok((namespace, packages, interfaces, function, version))
+                if packages.len() == 1
+                    && (interfaces.is_none()
+                        || interfaces.as_ref().is_some_and(|v| v.len() == 1)) =>
+            {
+                Ok(Self {
+                    namespace,
+                    package: packages
+                        .into_iter()
+                        .next()
+                        .context("unexpectedly missing package")?,
+                    interface: match interfaces {
+                        Some(v) => Some(
+                            v.into_iter()
+                                .next()
+                                .context("unexpectedly missing interface")?,
+                        ),
+                        None => None,
+                    },
+                    function,
+                    version,
+                })
+            }
+            Ok((_, _, _, Some(_), _)) => {
+                bail!("function-level interface overrides are not yet supported")
+            }
+            Ok(_) => bail!("nested interfaces not yet supported"),
+            Err(e) => bail!("failed to parse WIT interface spec (\"{s}\"): {e}"),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for WitInterfaceSpec {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(untagged)]
+        enum Multi {
+            Stringified(String),
+            Explicit {
+                namespace: String,
+                package: String,
+                interface: Option<String>,
+                function: Option<String>,
+                version: Option<Version>,
+            },
+        }
+
+        match Multi::deserialize(deserializer)? {
+            Multi::Stringified(s) => Self::from_str(&s).map_err(|e| {
+                serde::de::Error::custom(format!(
+                    "failed to parse WIT interface specification: {e}"
+                ))
+            }),
+            Multi::Explicit {
+                namespace,
+                package,
+                interface,
+                function,
+                version,
+            } => Ok(Self {
+                namespace,
+                package,
+                interface,
+                function,
+                version,
+            }),
+        }
+    }
+}
+
+/// Facilitates *one* of a given `T` or more (primarily for serde use)
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+#[serde(untagged)]
+pub enum OneOrMore<T> {
```

Comment:
I feel like this makes the understanding of the code a little harder, but I can be convinced if it really makes things simpler in this module

Reply:
Ah so the reason this is here is to enable a flow like in the RFC where you can provide *either* a simple object, or a list of them (for multiple kinds of things). This was a sort of "stretch goal" in the for example, if you want to override `wasi:keyvalue` most of the time you really have only *one* thing you want to override it with. 

But, you *could* have more than one. With `OneOrMore`, both specifying a TOML object and a TOML list work

Reply:
Makes sense! Thanks @vados-cosmonic